drivers/sensor/bh1745/bh1745.c

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2022 Grzegorz Blach
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <drivers/i2c.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <init.h>
+#include <kernel.h>
+#include <string.h>
+#include <logging/log.h>
+
+#include "bh1745.h"
+
+static struct k_work_delayable bh1745_init_work;
+static const struct device *bh1745_dev;
+
+LOG_MODULE_REGISTER(BH1745, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT rohm_bh1745
+
+static const int32_t async_init_delay[ASYNC_INIT_STEP_COUNT] = {
+	[ASYNC_INIT_STEP_RESET_CHECK] = 2,
+	[ASYNC_INIT_RGB_ENABLE] = 0,
+	[ASYNC_INIT_STEP_CONFIGURE] = 0
+};
+
+static int bh1745_async_init_reset_check(struct bh1745_data *data);
+static int bh1745_async_init_rgb_enable(struct bh1745_data *data);
+static int bh1745_async_init_configure(struct bh1745_data *data);
+
+static int(*const async_init_fn[ASYNC_INIT_STEP_COUNT])(
+	struct bh1745_data *data) = {
+	[ASYNC_INIT_STEP_RESET_CHECK] = bh1745_async_init_reset_check,
+	[ASYNC_INIT_RGB_ENABLE] = bh1745_async_init_rgb_enable,
+	[ASYNC_INIT_STEP_CONFIGURE] = bh1745_async_init_configure,
+};
+
+static int bh1745_sample_fetch(const struct device *dev, enum sensor_channel chan)
+
+{
+	struct bh1745_data *data = dev->data;
+	uint8_t status;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_ERR("Unsupported sensor channel");
+		return -ENOTSUP;
+	}
+
+	if (unlikely(!data->ready)) {
+		LOG_INF("Device is not initialized yet");
+		return -EBUSY;
+	}
+
+	LOG_DBG("Fetching sample...\n");
+
+	if (i2c_reg_read_byte(data->i2c, DT_REG_ADDR(DT_DRV_INST(0)),
```

Comment:
Please use the dt variant for all i2c transactions (`i2c_reg_read_byte_dt` and friends)

Reply:
Done